src/rest/SearchEndpoint.ts

```diff
@@ -1421,11 +1420,83 @@ function includeReferrer() {
   };
 }
 
+function includeClientId() {
```

Comment:
This function and the below ones are copy and pastes of the surrounding functions with only a different attribute. I tried simplifying those functions and experienced a very weird error. If you believe it's worth it, I could spend more time to simplify those functions and the surrounding ones.

Reply:
I think it's possible to create a function that contains the duplicated logic and accepts a string property name to build the correct thing. e.g.

```
const args = buildArgs('clientId');
return originalMethod.apply(this, args);
```
where
```
function buildArgs(key: keyof AnalyticsUtils) {
    ...
}
```

Reply:
@samisayegh I introduced the `getEndpointCallParameters` function and applied it pretty much everywhere possible. If you believe that this is a too big change, I can revert them.

Reply:
I think it is better, thanks.